internal/updater/install.go

```
@@ -0,0 +1,240 @@
+// runtime.GOOS string compares scattered through the package.
+//
+// Yes this is the same answer runtime.GOOS gives you; the build-
+// tagged variable form makes the test-only override trivial.
+
+// httpClient returns the underlying http.Client used by the github
+// releases client — install reuses it so TLS pooling / proxies /
+// timeouts are consistent with the poll path.
+//
+// Defined here (not in github.go) so tests in this package can build
+// a Client manually and still drive Install through it.
+func (c *Client) httpClient() *http.Client { return c.http }
```

golangci-lint (2.11.4)

[error] func (*Client).httpClient is unused

(unused)